recipe/meta.yaml

```
@@ -1,23 +1,23 @@
-{% set version = "3.0.6" %}
+{% set version = "4.0.1" %}
 {% set cuda_major = (cuda_compiler_version|default("12.0.0")).split(".")[0] %}
-{% set platform = "x86_64" %}  # [linux64]
-{% set platform = "SBSA" %}  # [aarch64]
-{% set platform = "windows" %}  # [win]
-{% set extension = "tgz" %}  # [not win]
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.gz" %}  # [not win]
 {% set extension = "zip" %}  # [win]
 
 package:
   name: nvcomp
   version: {{ version }}
 
 source:
-  - url: https://developer.download.nvidia.com/compute/nvcomp/{{ version }}/local_installers/nvcomp_{{ version }}_{{ platform }}_{{ cuda_major }}.x.{{ extension }}
-    sha256: 2631c4cb36c2b65e1c8d3a818bc470b849893e72bdbee0365d22dc3f7c130954  # [linux64 and (cuda_compiler_version or "None").startswith("11")]
-    sha256: a76511cc55a40aa9f8ca6b6b437102cc90969c69bfbcd66265f2e04f8f01435f  # [linux64 and (cuda_compiler_version or "None").startswith("12")]
-    sha256: 21b0b6ad262101a9d89bc5dc83cddf3da6b57b0a5ef8d75e5876e921ca46c9c5  # [aarch64 and (cuda_compiler_version or "None").startswith("11")]
-    sha256: d1a25754c21e37e643ebe5828ffdb11d0f237e27c4f29b83b5c17de2e44247d5  # [aarch64 and (cuda_compiler_version or "None").startswith("12")]
-    sha256: 10b98c8b58b9186656bfaf06ce8a7d06a5ce3c1318a9eaa254c47cadb71f748d  # [win and (cuda_compiler_version or "None").startswith("11")]
-    sha256: f7f25250d628e015854c0f65688921a2947f53f8b72aa01ce7c45aa2ea04893c  # [win and (cuda_compiler_version or "None").startswith("12")]
+  - url: https://developer.download.nvidia.com/compute/nvcomp/{{ version }}/local_installers/nvcomp-{{ platform }}-{{ version }}-cuda{{ cuda_major }}.x.{{ extension }}
```

Thanks Bradley! 🙏

Seeing this failure in [the re-rendering bot]( https://github.com/conda-forge/nvcomp-feedstock/actions/runs/10584940355/job/29330373999#step:3:605 )

```
jinja2.exceptions.UndefinedError: 'platform' is undefined
```

Think that is because `platform` is conditionally defined above. Believe this coming up now due to a change in conda-build 24.7 regarding handling of undefined Jinja values in `skip`ped recipes ( https://github.com/conda-forge/conda-smithy/issues/2011 ). Michael has a fix for that in PR: https://github.com/conda/conda-build/pull/5458

For now, think we can bypass this by similarly conditioning this URL so that it is always using defined `platform`s, re-rendering should work again

```suggestion
  - url: https://developer.download.nvidia.com/compute/nvcomp/{{ version }}/local_installers/nvcomp-{{ platform }}-{{ version }}-cuda{{ cuda_major }}.x.{{ extension }}  # [linux64 or aarch64 or win]
```